xmlserializer/XmlStreamDocSink.cpp

```diff
@@ -27,36 +27,34 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-#include "XmlStringDocSink.h"
+#include "XmlStreamDocSink.h"
 #include <libxml/parser.h>
 
 #define base CXmlDocSink
 
-CXmlStringDocSink::CXmlStringDocSink(std::string& strResult):
-      _strResult(strResult)
+CXmlStreamDocSink::CXmlStreamDocSink(std::ostream& output):
+    _output(output)
 {
 }
 
-bool CXmlStringDocSink::doProcess(CXmlDocSource& xmlDocSource,
+bool CXmlStreamDocSink::doProcess(CXmlDocSource& xmlDocSource,
                                   CXmlSerializingContext& serializingContext)
 {
-    (void)serializingContext;
-
-    xmlChar* pcDumpedDoc = NULL;
+    xmlChar* dumpedDoc = NULL;
 
     int iSize;
-    xmlDocDumpFormatMemoryEnc(xmlDocSource.getDoc(), &pcDumpedDoc, &iSize, "UTF-8", 1);
+    xmlDocDumpFormatMemoryEnc(xmlDocSource.getDoc(), &dumpedDoc, &iSize, "UTF-8", 1);
 
-    if (!pcDumpedDoc) {
+    if (!dumpedDoc) {
 
         serializingContext.setError("Unable to encode XML document in memory");
 
         return false;
     }
 
-    _strResult.append((const char*)pcDumpedDoc);
+    _output << (const char*)dumpedDoc;
```

Comment:
Are you sure you can safely cast `xmlChar*` to `const char *`. Maybe use a static_cast? I locked at the doc, it seems xmlChar is a typedef of unsined char, as a result no cast seems needed.